control/autoware_lane_departure_checker/src/lane_departure_checker_node/lane_departure_checker.cpp

```diff
@@ -386,6 +403,37 @@ bool LaneDepartureChecker::checkPathWillLeaveLane(
     });
 }
 
+bool LaneDepartureChecker::checkPathWillLeaveLane(
+  const lanelet::LaneletMapPtr lanelet_map_ptr, const PathWithLaneId & path,
+  std::vector<lanelet::Id> & fused_lanelets_id,
+  std::optional<autoware::universe_utils::Polygon2d> & fused_lanelets_polygon) const
+{
+  universe_utils::ScopedTimeTrack st(__func__, *time_keeper_);
+
+  // check if the footprint is not fully contained within the fused lanelets polygon
+  const std::vector<LinearRing2d> vehicle_footprints = createVehicleFootprints(path);
+
+  if (fused_lanelets_polygon) {
+    bool is_inside_fused_lanelets = std::all_of(
+      vehicle_footprints.begin(), vehicle_footprints.end(), [&](const auto & footprint) {
+        return boost::geometry::within(footprint, fused_lanelets_polygon.value());
+      });
+
+    if (is_inside_fused_lanelets) {
+      return false;
+    }
+  }
+
+  const auto updated = updateFusedLaneletPolygonForPath(
+    lanelet_map_ptr, path, fused_lanelets_id, fused_lanelets_polygon);
+  if (!updated) return true;
+
+  return !std::all_of(
+    vehicle_footprints.begin(), vehicle_footprints.end(), [&](const auto & footprint) {
+      return boost::geometry::within(footprint, fused_lanelets_polygon.value());
+    });
```

Comment:
```suggestion
  const std::vector<LinearRing2d> vehicle_footprints = createVehicleFootprints(path);

  auto is_all_footprints_within = [&](const auto & polygon) {
    return std::all_of(
      vehicle_footprints.begin(), vehicle_footprints.end(),
      [&polygon](const auto & footprint) { return boost::geometry::within(footprint, polygon); });
  };

  // If lanelets polygon exists and all footprints are within it, the path doesn't leave the lane
  if (fused_lanelets_polygon && is_all_footprints_within(fused_lanelets_polygon.value())) {
    return false;
  }

  // Update the lanelet polygon for the current path
  if (!updateFusedLaneletPolygonForPath(
        lanelet_map_ptr, path, fused_lanelets_id, fused_lanelets_polygon)) {
    // If update fails, assume the path leaves the lane
    return true;
  }

  // Check if any footprint is outside the updated lanelets polygon
  return !is_all_footprints_within(fused_lanelets_polygon.value());
```
It's a matter of preference, but what do you think about the following modification?